validator.js

```diff
@@ -450,7 +450,26 @@
     };
 
     validator.isDate = function (str) {
-        return !isNaN(Date.parse(str));
+        var rawDate = new Date(str);
+        var normalizedDate = new Date(rawDate.getTime() + rawDate.getTimezoneOffset()*60000);
```

Comment:
A valid Date object like `new Date('2011-09-30')` could be interpreted differently depending on the javascript engine.

In webkit browsers and node.js, this returns `Thu Sep 29 2011 20:00:00 GMT-0400 (EDT)`
whereas in Firefox, this returns `2011-09-30 T00:00:00.000Z`. I.e., different days are interpreted. Adding `getTimezoneOffset()` standardizes the date.